proto/noble/dollar/vaults/v2/tx.proto

```diff
@@ -0,0 +1,688 @@
+syntax = "proto3";
+
+package noble.dollar.vaults.v2;
+
+import "amino/amino.proto";
+import "cosmos/msg/v1/msg.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+import "noble/dollar/vaults/v1/vaults.proto";
+import "noble/dollar/vaults/v2/cross_chain.proto";
+import "noble/dollar/vaults/v2/genesis.proto";
+import "noble/dollar/vaults/v2/oracle.proto";
+
+option go_package = "dollar.noble.xyz/v2/types/vaults/v2";
+
+// Msg defines the V2 vaults transaction service
+service Msg {
+  option (cosmos.msg.v1.service) = true;
+
+  // Deposit into a V2 vault
+  rpc Deposit(MsgDeposit) returns (MsgDepositResponse);
+
+  // Request withdrawal (queue-based)
+  rpc RequestWithdrawal(MsgRequestWithdrawal) returns (MsgRequestWithdrawalResponse);
+
+  // Set yield preference for a user's position
+  rpc SetYieldPreference(MsgSetYieldPreference) returns (MsgSetYieldPreferenceResponse);
```

Comment:
There are several message handlers with messages that are not in the spec, I'll make note of them. This is one.